File: Top 100 Codes/06-Advanced Codes related to Arrays/27-Minimum no. of operations required to make an array palindrome/Minimumnumofoperationsrequiredtomakeanarraypalindrome.cpp
#include <iostream>
#include <vector>
using namespace std;

// Function to find minimum operations to make array palindrome
int minOperationsToMakePalindrome(vector<int>& arr) {
    int i = 0, j = arr.size() - 1;
    int operations = 0;

    while (i < j) {
        if (arr[i] == arr[j]) {
            i++;
            j--;
        } else if (arr[i] < arr[j]) {
            arr[i + 1] += arr[i];
            i++;
            operations++;
        } else {
            arr[j - 1] += arr[j];
            j--;
            operations++;
        }
    }

    return operations;
}

int main() {
    vector<int> arr = {1, 4, 5, 9, 1};
    cout << "Minimum operations to make palindrome: " << minOperationsToMakePalindrome(arr) << endl;
    return 0;
}

File: Top 100 Codes/06-Advanced Codes related to Arrays/27-Minimum no. of operations required to make an array palindrome/Minimumnumofoperationsrequiredtomakeanarraypalindrome.py
def min_operations_to_make_palindrome(arr):
    i = 0
    j = len(arr) - 1
    operations = 0

    while i < j:
        if arr[i] == arr[j]:
            i += 1
            j -= 1
        elif arr[i] < arr[j]:
            arr[i + 1] += arr[i]
            i += 1
            operations += 1
        else:
            arr[j - 1] += arr[j]
            j -= 1
            operations += 1

    return operations

# Example usage
arr = [1, 4, 5, 9, 1]
print("Minimum operations to make palindrome:", min_operations_to_make_palindrome(arr))
